core/dbt/contracts/graph/nodes.py

```diff
@@ -241,10 +245,77 @@ class ParsedNode(NodeInfoMixin, ParsedNodeMandatory, SerializableType):
     unrendered_config: Dict[str, Any] = field(default_factory=dict)
     created_at: float = field(default_factory=lambda: time.time())
     config_call_dict: Dict[str, Any] = field(default_factory=dict)
+    constraints_enabled: bool = False
     relation_name: Optional[str] = None
     raw_code: str = ""
 
+    def constraints_validator_handler(self):
```

Comment:
this might be cleaner if we moved it to it's own file, maybe `contracts/graph/contracts.py`? I'll defer to others with more experience in this part of the code